src/Files.App.Controls/ThemedIcon/ThemedIcon.cs

```diff
@@ -6,371 +6,224 @@
-                }
-                else
-                {
-                    VisualStateManager.GoToState(this, OutlineTypeStateName, true);
-                }
-            }
-
-            VisualStateManager.GoToState(this, EnabledStateName, true);
-        }
-
-        private void UpdateIconColorTypeStates()
-        {
-            /*
-            // Handles changes to the IconColorType and setting the correct Visual States.
-
-            // We first check if the Icon is Disabled
-            // Then we check if the Disabled Icon is Toggled
-
-            // We then assume the Icon is Enabled
-            // We then check the Toggled state for the Contrast Icons
-            // We have two states depending on toggle.
-
-            // Finally we act on all other Enabled states
-            // We check for Toggled state
-            // And update the IconColorType in the Layered Icon's Layers
-            */
-
-            if (_isEnabled is false || IsEnabled is false)
-            {
-                if (_isToggled is true || IsToggled is true)
-                {
-                    VisualStateManager.GoToState(this, DisabledToggleColorStateName, true);
-                }
-                else
-                {
-                    VisualStateManager.GoToState(this, DisabledColorStateName, true);
-                }
-            }
-            else
-            {
-                if (_isToggled is true || IsToggled is true)
-                {
-                    VisualStateManager.GoToState(this, ToggleStateName, true);
-                }
-                else
-                {
-                    VisualStateManager.GoToState(
-                        this,
-                        IconColorType switch
-                        {
-                            ThemedIconColorType.Critical => CriticalStateName,
-                            ThemedIconColorType.Caution => CautionStateName,
-                            ThemedIconColorType.Success => SuccessStateName,
-                            ThemedIconColorType.Neutral => NeutralStateName,
-                            ThemedIconColorType.Accent => AccentStateName,
-                            ThemedIconColorType.Custom => CustomColorStateName,
-                            _ => NormalStateName,
-                        },
-                        true);
-                }
-
-                if (GetTemplateChild(LayeredPathCanvas) is Canvas canvas)
-                {
-                    foreach (var layer in canvas.Children.Cast<ThemedIconLayer>())
-                        layer.IconColorType = IconColorType;
-                }
-            }
-        }
-    }
+	/// <summary>
+	/// A control for a State and Color aware Icon
+	/// </summary>
+	public partial class ThemedIcon : Control
+	{
+		public ThemedIcon()
+		{
+			DefaultStyleKey = typeof(ThemedIcon);
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			IsEnabledChanged -= OnIsEnabledChanged;
```

Comment:
Why are we unregistering before applying template?

Reply:
I have seen other controls handle their event registry in this way, I assume its to have a fresh state when the Template is applied

Reply:
I don't think this is actually necessary given that the template does not have to do anything with your own event.

Reply:
I will remove the unregistration